src/main/java/core/basesyntax/service/impl/ReaderFromFileImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReaderFromFile;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class ReaderFromFileImpl implements ReaderFromFile {
+    @Override
+    public String[] readFromFile(String fileName) {
+        File file = new File(fileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String value = reader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = reader.readLine();
+            }
```

Comment:
You create a huge String and then split it to an array.
May be better return List<String>? You can get rid of StringBuilder and simply add elements to List inside your while loop.